From 35e65650b81464d98211d32f6e6dcaca7df25ad6 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Wed, 5 Sep 2012 14:33:00 +0800
Subject: [PATCH 335/502] ENGR00221840-2 jb upgrade: Update the hardware
 module structure define

Remove the const type for hardware module define
due to module load will overwrite the dso member.
Update for GPS and alsa.

Change-Id: I40b9067475739f635ad093032ee39dada30b7568
Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
Signed-off-by: jiejing.zhang <jiejing.zhang@freescale.com>
---
 alsa/alsa_imx.cpp   |  2 +-
 alsa/tinyalsa_hal.c |  2 +-
 libgps/gps.c        | 17 +++++++++--------
 3 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/alsa/alsa_imx.cpp b/alsa/alsa_imx.cpp
index b171842..2554fae 100755
--- a/alsa/alsa_imx.cpp
+++ b/alsa/alsa_imx.cpp
@@ -85,7 +85,7 @@ static hw_module_methods_t s_module_methods = {
     open            : s_device_open
 };
 
-extern "C" const hw_module_t HAL_MODULE_INFO_SYM = {
+extern "C" hw_module_t HAL_MODULE_INFO_SYM = {
     tag             : HARDWARE_MODULE_TAG,
     version_major   : 1,
     version_minor   : 0,
diff --git a/alsa/tinyalsa_hal.c b/alsa/tinyalsa_hal.c
index 1f148b8..539f796 100644
--- a/alsa/tinyalsa_hal.c
+++ b/alsa/tinyalsa_hal.c
@@ -2064,7 +2064,7 @@ static int adev_open(const hw_module_t* module, const char* name,
         return -ENOMEM;
 
     adev->hw_device.common.tag      = HARDWARE_DEVICE_TAG;
-    adev->hw_device.common.version  = 0;
+    adev->hw_device.common.version  = AUDIO_DEVICE_API_VERSION_1_0;
     adev->hw_device.common.module   = (struct hw_module_t *) module;
     adev->hw_device.common.close    = adev_close;
 
diff --git a/libgps/gps.c b/libgps/gps.c
index 1c7585d..176c19c 100644
--- a/libgps/gps.c
+++ b/libgps/gps.c
@@ -32,6 +32,7 @@
   *
   ***************************************************************************/
 
+#include <hardware/hardware.h>
 #include <hardware/gps.h>
 
 #include <stdlib.h>
@@ -63,12 +64,12 @@ static struct hw_module_methods_t gps_module_methods = {
     .open = open_gps
 };
 
-const struct hw_module_t HAL_MODULE_INFO_SYM = {
-    .tag = HARDWARE_MODULE_TAG,
-    .version_major = 1,
-    .version_minor = 0,
-    .id = GPS_HARDWARE_MODULE_ID,
-    .name = "Atheros GPS Module",
-    .author = "FSL MAD, Inc.",
-    .methods = &gps_module_methods,
+struct hw_module_t HAL_MODULE_INFO_SYM = {
+	.tag = HARDWARE_MODULE_TAG,
+	.version_major = 1,
+	.version_minor = 0,
+	.id = GPS_HARDWARE_MODULE_ID,
+	.name = "Atheros GPS Module",
+	.author = "FSL MAD, Inc.",
+	.methods = &gps_module_methods,
 };
-- 
1.8.0

